x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -26,108 +25,99 @@ import (
 //
 // Note: for now, to get proofs in your ICQs, you need to query the entire store on the host zone! e.g. "store/bank/key"
 func DelegatorSharesCallback(k Keeper, ctx sdk.Context, args []byte, query icqtypes.Query) error {
-	hostZone, found := k.GetHostZone(ctx, query.GetChainId())
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(query.ChainId, ICQCallbackID_Delegation,
+		"Starting delegator shares callback, QueryId: %vs, QueryType: %s, Connection: %s", query.Id, query.QueryType, query.ConnectionId))
+
+	// Confirm host exists
+	chainId := query.ChainId
+	hostZone, found := k.GetHostZone(ctx, chainId)
 	if !found {
-		errMsg := fmt.Sprintf("no registered zone for queried chain ID (%s)", query.GetChainId())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrHostZoneNotFound, errMsg)
+		return sdkerrors.Wrapf(types.ErrHostZoneNotFound, "no registered zone for queried chain ID (%s)", chainId)
 	}
 
 	// Unmarshal the query response which returns a delegation object for the delegator/validator pair
 	queriedDelgation := stakingtypes.Delegation{}
 	err := k.cdc.Unmarshal(args, &queriedDelgation)
 	if err != nil {
-		errMsg := fmt.Sprintf("unable to unmarshal queried delegation info for zone %s, err: %s", hostZone.ChainId, err.Error())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrMarshalFailure, errMsg)
+		return sdkerrors.Wrapf(types.ErrMarshalFailure, "unable to unmarshal query response into Delegation type, err: %s", err.Error())
 	}
-	k.Logger(ctx).Info(fmt.Sprintf("DelegationCallback: HostZone: %s, Delegator: %s, Validator: %s, Shares: %v",
-		hostZone.ChainId, queriedDelgation.DelegatorAddress, queriedDelgation.ValidatorAddress, queriedDelgation.Shares))
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Delegation, "Query response - Delegator: %s, Validator: %s, Shares: %v",
+		queriedDelgation.DelegatorAddress, queriedDelgation.ValidatorAddress, queriedDelgation.Shares))
 
-	// ensure ICQ can be issued now! else fail the callback
-	isWithinWindow, err := k.IsWithinBufferWindow(ctx)
+	// Ensure ICQ can be issued now, else fail the callback
+	withinBufferWindow, err := k.IsWithinBufferWindow(ctx)
 	if err != nil {
-		errMsg := fmt.Sprintf("unable to determine if ICQ callback is inside buffer window, err: %s", err.Error())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrOutsideIcqWindow, errMsg)
-	} else if !isWithinWindow {
-		k.Logger(ctx).Error("delegator shares callback is outside ICQ window")
-		return nil
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "unable to determine if ICQ callback is inside buffer window, err: %s", err.Error())
+	}
+	if !withinBufferWindow {
+		// QUESTION: Should this return an error?
```

Comment:
Returning a failure here makes the caller aware they're outside the buffer window, and tells them how far outside it they are. It's not necessary, just convenient.

Reply:
Cool, I just noticed some of the ICQ functions returned an error when we were outside the buffer window and other's did not (including this one). So wanted to make sure there wasn't a specific reason for not returning an error in these